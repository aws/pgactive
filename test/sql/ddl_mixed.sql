-- test for RT-#37869

CREATE TABLE add_column (
    id serial primary key,
    data text
);

INSERT INTO add_column (data) SELECT generate_series(1,100,10);

ALTER TABLE add_column ADD COLUMN other varchar(100);

SELECT bdr.bdr_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
\c postgres
SELECT id, data, other FROM add_column ORDER BY id;

UPDATE add_column SET other = 'foobar';

SELECT bdr.bdr_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
\c regression
SELECT id, data, other FROM add_column ORDER BY id;

DROP TABLE add_column;

-- We allow BDR nodes sending out changes for postgres logical replication
-- subscribers.
CREATE PUBLICATION mypub FOR ALL TABLES;

-- We do not allow BDR nodes receiving changes from postgres logical
-- replication publishers.
CREATE SUBSCRIPTION mysub CONNECTION '' PUBLICATION mypub;
ALTER SUBSCRIPTION mysub REFRESH PUBLICATION;

DROP PUBLICATION mypub;

-- We do not allow external logical replication extensions to be created when
-- BDR is active.
CREATE EXTENSION pglogical;
