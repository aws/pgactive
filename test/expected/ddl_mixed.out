-- test for RT-#37869
CREATE TABLE add_column (
    id serial primary key,
    data text
);
INSERT INTO add_column (data) SELECT generate_series(1,100,10);
ALTER TABLE add_column ADD COLUMN other varchar(100);
SELECT pgactive.pgactive_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
 pgactive_wait_for_slots_confirmed_flush_lsn 
---------------------------------------------
 
(1 row)

\c postgres
SELECT id, data, other FROM add_column ORDER BY id;
 id | data | other 
----+------+-------
  1 | 1    | 
  2 | 11   | 
  3 | 21   | 
  4 | 31   | 
  5 | 41   | 
  6 | 51   | 
  7 | 61   | 
  8 | 71   | 
  9 | 81   | 
 10 | 91   | 
(10 rows)

UPDATE add_column SET other = 'foobar';
SELECT pgactive.pgactive_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
 pgactive_wait_for_slots_confirmed_flush_lsn 
---------------------------------------------
 
(1 row)

\c regression
SELECT id, data, other FROM add_column ORDER BY id;
 id | data | other  
----+------+--------
  1 | 1    | foobar
  2 | 11   | foobar
  3 | 21   | foobar
  4 | 31   | foobar
  5 | 41   | foobar
  6 | 51   | foobar
  7 | 61   | foobar
  8 | 71   | foobar
  9 | 81   | foobar
 10 | 91   | foobar
(10 rows)

DROP TABLE add_column;
-- We allow pgactive nodes sending out changes for postgres logical replication
-- subscribers.
CREATE PUBLICATION mypub FOR ALL TABLES;
-- We do not allow pgactive nodes receiving changes from postgres logical
-- replication publishers.
CREATE SUBSCRIPTION mysub CONNECTION '' PUBLICATION mypub;
ERROR:  CREATE SUBSCRIPTION is not supported when pgactive is active
ALTER SUBSCRIPTION mysub REFRESH PUBLICATION;
ERROR:  ALTER SUBSCRIPTION is not supported when pgactive is active
DROP PUBLICATION mypub;
-- We do not allow external logical replication extensions to be created when
-- pgactive is active.
CREATE EXTENSION pglogical;
ERROR:  cannot create an external logical replication extension when pgactive is active
